File: app/src/main/java/shts/jp/android/nogifeed/activities/OtherMenuActivity.java
package shts.jp.android.nogifeed.activities;

import android.support.v4.app.FragmentTransaction;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import shts.jp.android.nogifeed.R;
import shts.jp.android.nogifeed.fragments.AboutFragment;
import shts.jp.android.nogifeed.fragments.LisencesFragment;
import shts.jp.android.nogifeed.fragments.RequestFragment;

/**
 * ドロワーのそのほかメニューから遷移する画面
 */
public class OtherMenuActivity extends AppCompatActivity {

    private static final String TAG = OtherMenuActivity.class.getSimpleName();

    public static Intent getStartIntent(Context context, int id) {
        Intent intent = new Intent(context, OtherMenuActivity.class);
        intent.putExtra("menu_id", id);
        return intent;
    }

    private final Fragment getFragmentFrom(int id) {
        switch (id) {
            case R.id.menu_lisences : return new LisencesFragment();
            case R.id.menu_request : return new RequestFragment();
            case R.id.menu_about_app : return new AboutFragment();
        }
        return null;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_common);

        Fragment fragment = getFragmentFrom(getIntent().getIntExtra("menu_id", -1));
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.container, fragment, OtherMenuActivity.class.getSimpleName());
        ft.commit();
    }
}

File: app/src/main/java/shts/jp/android/nogifeed/activities/OtherMenuActivity.kt
package shts.jp.android.nogifeed.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import shts.jp.android.nogifeed.R
import shts.jp.android.nogifeed.fragments.AboutFragment
import shts.jp.android.nogifeed.fragments.LicenseFragment
import shts.jp.android.nogifeed.fragments.RequestFragment

/**
 * ドロワーのそのほかメニューから遷移する画面
 */
class OtherMenuActivity : AppCompatActivity() {

    companion object {
        private val EXTRA_MENU_ID = "menu_id"

        fun getStartIntent(context: Context, id: Int): Intent {
            val intent = Intent(context, OtherMenuActivity::class.java)
            intent.putExtra(EXTRA_MENU_ID, id)
            return intent
        }
    }

    private fun getFragmentFrom(id: Int): Fragment? {
        when (id) {
            R.id.menu_lisences -> return LicenseFragment()
            R.id.menu_request -> return RequestFragment()
            R.id.menu_about_app -> return AboutFragment()
        }
        return null
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_common)

        val fragment = getFragmentFrom(intent.getIntExtra(EXTRA_MENU_ID, -1))
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.container, fragment, OtherMenuActivity::class.java.simpleName)
        ft.commit()
    }
}
